cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -100,19 +109,68 @@ pub enum Error {
     Network(#[from] network::Error),
     #[error(transparent)]
     Wasm(#[from] wasm::Error),
+    #[error("alias \"{alias}\" already exist. Use --force to override it.")]
+    AliasAlreadyExist { alias: String },
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
+        self.validate_alias()?;
+
         let res = self.run_against_rpc_server(None, None).await?.to_envelope();
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
+                self.save_contract_id(&contract);
                 println!("{contract}");
             }
         }
         Ok(())
     }
+
+    fn validate_alias(&self) -> Result<(), Error> {
+        let alias = self.alias();
+
+        if alias.is_empty() {
+            return Ok(());
+        }
+
+        let path = self.alias_path();
+
+        if path.exists() && !self.force {
+            Err(Error::AliasAlreadyExist { alias })
+        } else {
+            Ok(())
+        }
+    }
+
+    fn alias(&self) -> String {
+        match &self.alias {
+            Some(n) => n.to_string(),
+            None => String::new(),
+        }
+    }
+
+    fn alias_path(&self) -> PathBuf {
+        let config_dir = self.config.config_dir().unwrap();
+        let name = format!("{}.txt", self.alias());
+
+        config_dir.join("contract-ids").join(name)
+    }
+
+    fn save_contract_id(&self, contract: &String) {
+        let file_path = self.alias_path();
+        let dir = file_path.parent().unwrap();
+
+        let _ = create_dir_all(dir);
+
+        let mut to_file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .open(file_path)
+            .unwrap();
+        let _ = to_file.write_all(contract.as_bytes());
```

Comment:
I would also avoid it being possible to assign the same alias multiple times with different networks, and instead take the same approach that keys has that the alias "id space" is across all networks, just so that it is less confusing and to remove the ability to have multiple values assigned to a single alias. But something that is different for aliases than keys, is that contract IDs will never be the same across networks, so now or in the future if we store the network passphrase (not network name) in the .json, we should error if someone is trying to use an alias on the wrong network because we're just saving them from doing stuff that won't work.